test/e2e/virtual/apiexport/authorizer_test.go

```diff
@@ -98,131 +95,198 @@ func TestAPIExportAuthorizers(t *testing.T) {
+		&apisv1alpha1.APIBinding{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "wild.wild.west",
+			},
+			Spec: apisv1alpha1.APIBindingSpec{
+				Reference: apisv1alpha1.BindingReference{
+					Export: &apisv1alpha1.ExportBindingReference{
+						Path: serviceProvider1Path.String(),
+						Name: "wild.wild.west",
+					},
 				},
 			},
 		},
-	}
-	framework.Eventually(t, func() (bool, string) {
-		_, err = user3KcpClient.Cluster(tenantPath).ApisV1alpha1().APIBindings().Create(ctx, apiBinding, metav1.CreateOptions{})
-		if err != nil {
-			return false, fmt.Sprintf("error creating API binding: %v", err)
-		}
-		return true, ""
-	}, wait.ForeverTestTimeout, time.Millisecond*100, "api binding creation failed")
+		&apisv1alpha1.APIBinding{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "cowboys",
+			},
+			Spec: apisv1alpha1.APIBindingSpec{
+				PermissionClaims: []apisv1alpha1.AcceptablePermissionClaim{
+					{
+						PermissionClaim: apisv1alpha1.PermissionClaim{
+							GroupResource: apisv1alpha1.GroupResource{Resource: "configmaps"},
+							ResourceSelector: []apisv1alpha1.ResourceSelector{
+								{Name: "claimed-cm", Namespace: "claimed-ns-by-serviceprovider-2"},
+								{Name: "claimed-deadlock-cm", Namespace: "deadlock-cm-namespace"},
+							},
+						},
+						State: apisv1alpha1.ClaimAccepted,
+					},
+					{
+						PermissionClaim: apisv1alpha1.PermissionClaim{
+							GroupResource: apisv1alpha1.GroupResource{Group: "wild.wild.west", Resource: "sheriffs"},
+							IdentityHash:  sherriffsIdentityHash,
+							All:           true,
+						},
+						State: apisv1alpha1.ClaimAccepted,
+					},
+				},
+				Reference: apisv1alpha1.BindingReference{
+					Export: &apisv1alpha1.ExportBindingReference{
+						Path: serviceProvider2Path.String(),
+						Name: "today-cowboys",
+					},
+				},
+			},
+		},
+	))
 
 	t.Logf("Make sure [%q, %q] API groups shows up in consumer workspace %q group discovery", wildwest.GroupName, "wild.wild.west", tenantPath)
 	user3tenantWorkspaceKcpClient, err := kcpclientset.NewForConfig(framework.UserConfig("user-3", rest.CopyConfig(cfg)))
 	require.NoError(t, err)
-	err = wait.PollImmediateWithContext(ctx, 100*time.Millisecond, wait.ForeverTestTimeout, func(c context.Context) (done bool, err error) {
+	framework.Eventually(t, func() (success bool, reason string) {
 		groups, err := user3tenantWorkspaceKcpClient.Cluster(tenantPath).Discovery().ServerGroups()
 		if err != nil {
-			return false, fmt.Errorf("error retrieving consumer workspace %q group discovery: %w", tenantPath, err)
+			return false, fmt.Sprintf("error retrieving consumer workspace %q group discovery: %v", tenantPath, err)
 		}
-		return groupExists(groups, wildwest.GroupName) && groupExists(groups, "wild.wild.west"), nil
-	})
-	require.NoError(t, err)
+		return groupExists(groups, wildwest.GroupName) && groupExists(groups, "wild.wild.west"), ""
+	}, wait.ForeverTestTimeout, 100*time.Millisecond, "listing claimed resources failed")
```

Comment:
nit: we're calling discovery, not a LIST

Reply:
fixed